src/Equinox.CosmosStore/CosmosStore.fs

```diff
@@ -625,7 +625,8 @@ module internal Query =
         let mutable i = 0
         while query.HasMoreResults do
             let! t, (res: FeedResponse<'t>) = (fun ct -> query.ReadNextAsync(ct)) |> Stopwatch.time ct
-            yield map i t res }
+            yield map i t res
+            i <- i + 1 }
```

Comment:
this fixes values in the logs, and makes the maxbatches limit be enforced correctly